src/components/DateInput/DateInput.tsx

```diff
@@ -77,7 +85,19 @@ const DateInput: React.FC<DateInputProps & Omit<TextInputProps, 'value' | 'onCha
   const targetRef = React.useRef(null);
   const [currentDate, setCurrentDate] = useState(dateToDayjs(value, timezone));
   const [currentMonth, setCurrentMonth] = useState(currentDate || now(timezone));
+  const resetLabel = React.useMemo(() => (withTime ? 'Clear Date & Time' : 'Clear Date'), [
+    withTime,
+  ]);
+
   const { isOpen, open, close } = useDisclosure();
+  const previousDate = usePrevious(dateToDayjs(value, timezone));
```

Comment:
Wouldn't that be more correct?

```suggestion
  const previousDate = usePrevious(currentDate);
```

Reply:
Same as above, we need to check the controlled value not the currentDate as it is changing.